networking_cisco/plugins/cisco/cfg_agent/service_helpers/routing_svc_helper_aci.py

```diff
@@ -77,8 +77,18 @@ def _process_gateway_set(self, ri, ex_gw_port, list_port_ids_up):
         self._process_new_ports(ri, new_ports, ex_gw_port, list_port_ids_up)
 
     def _process_gateway_cleared(self, ri, ex_gw_port):
-        super(RoutingServiceHelperAci,
-              self)._process_gateway_cleared(ri, ex_gw_port)
+        # We're about to remove this router ID from the list,
+        # so check if it's the last router for this VRF. If it is, we can
+        # call to remove the external gateway.
+        if ri.ex_gw_port or ri.router.get('gw_port'):
+            driver = self.driver_manager.get_driver(ri.id)
+            vrf_name = driver._get_vrf_name(ri)
+            if self._router_ids_by_vrf.get(vrf_name) and (
```

Comment:
In this block of code we are calling this API:
self._router_ids_by_vrf.get(vrf_name)
3 times which is not efficient I think. Can we do something like this:

router_ids = self._router_ids_by_vrf.get(vrf_name)
if router_ids and ri.router['id'] in router_ids and len(router_ids) == 1:
    self._external_gateway_removed(ri, ex_gw_port)